tests/cluster/cluster.go

```diff
@@ -0,0 +1,181 @@
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package cluster
+
+import (
+	"fmt"
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	restclient "k8s.io/client-go/rest"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	netclient "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1"
+	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+
+	"github.com/kubevirt/ovs-cni/tests/node"
+)
+
+var Kubeconfig *string
+var RestConfig *restclient.Config
+var Clientset *kubernetes.Clientset
+var NetClient *netclient.K8sCniCncfIoV1Client
+var TestNamespace = "test-namespace"
+
+func CreateTestNamespace() error {
+	By(fmt.Sprintf("creating %s namespace", TestNamespace))
+	_, err := Clientset.CoreV1().Namespaces().Create(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: TestNamespace}}, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to create test namespace")
+	}
+	return nil
+}
+
+func RemoveTestNamespace() error {
+	By(fmt.Sprintf("Waiting for namespace %s to be removed, this can take a while ...\n", TestNamespace))
+	err := Clientset.CoreV1().Namespaces().Delete(context.TODO(), TestNamespace, metav1.DeleteOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to delete test namespace")
+	}
+
+	EventuallyWithOffset(1, func() error {
+		_, err := Clientset.CoreV1().Namespaces().Get(context.TODO(), TestNamespace, metav1.GetOptions{})
+		return err
+	}, 120*time.Second, 5*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(apierrors.IsNotFound, BeTrue())), "Should succeed terminating the namespace")
+
+	return nil
+}
+
+func CreatePrivilegedPodWithIp(podName, nadName, cidr string) error {
+	By(fmt.Sprintf("Creating pod %s with priviliged premission and ip %s", podName, cidr))
+	privileged := true
+	podObject := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{
+		Name: podName,
+		Namespace: TestNamespace,
+		// This annotation makes sure the pod is assigned to a node that has this ovs bridge resource
+		Annotations: map[string]string{"k8s.v1.cni.cncf.io/networks": nadName},
+	},
+	Spec: corev1.PodSpec{ Containers: []corev1.Container{{Name: "test",
+		Image:   "alpine",
+		Command: []string{"sh", "-c", fmt.Sprintf("ip address add %s dev net1; sleep 99999", cidr)},
+		SecurityContext: &corev1.SecurityContext{Privileged: &privileged}}}}}
+
+	_, err := Clientset.CoreV1().Pods(TestNamespace).Create(context.TODO(), podObject, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to wait for pod container to be in Running state")
+	}
+
+	By("waiting for pod container to be in Running state")
+	Eventually(func() bool {
```

Comment:
If this is used only in BeforeEach, we don't need to return issues and stick to `Eventually` and `Expect`. I believe that would be valid.

If there is an util that is used within `It`, it should not use `Gomega` asserters and return an error instead.

Reply:
ack.
fixed throughout the test.
Done